slither/utils/upgradeability.py

```diff
@@ -0,0 +1,463 @@
+from typing import Optional
+from slither.core.declarations import (
+    Contract,
+    Structure,
+    Enum,
+    SolidityVariableComposed,
+    SolidityVariable,
+    Function,
+)
+from slither.core.solidity_types import (
+    ElementaryType,
+    ArrayType,
+    MappingType,
+    UserDefinedType,
+)
+from slither.core.variables.local_variable import LocalVariable
+from slither.core.variables.local_variable_init_from_tuple import LocalVariableInitFromTuple
+from slither.core.variables.state_variable import StateVariable
+from slither.analyses.data_dependency.data_dependency import get_dependencies
+from slither.core.variables.variable import Variable
+from slither.core.expressions.literal import Literal
+from slither.core.expressions.identifier import Identifier
+from slither.core.expressions.call_expression import CallExpression
+from slither.core.expressions.assignment_operation import AssignmentOperation
+from slither.core.cfg.node import Node, NodeType
+from slither.slithir.operations import (
+    Assignment,
+    Index,
+    Member,
+    Length,
+    Binary,
+    Unary,
+    Condition,
+    NewArray,
+    NewStructure,
+    NewContract,
+    NewElementaryType,
+    SolidityCall,
+    Delete,
+    EventCall,
+    LibraryCall,
+    InternalDynamicCall,
+    HighLevelCall,
+    LowLevelCall,
+    TypeConversion,
+    Return,
+    Transfer,
+    Send,
+    Unpack,
+    InitArray,
+    InternalCall,
+)
+from slither.slithir.variables import (
+    TemporaryVariable,
+    TupleVariable,
+    Constant,
+    ReferenceVariable,
+)
+from slither.tools.read_storage.read_storage import SlotInfo, SlitherReadStorage
+
+
+# pylint: disable=too-many-locals
+def compare(v1: Contract, v2: Contract) -> dict:
```

Comment:
Should we return a dict here? Maybe we can return the 6 list directly?

Using a dict will make it difficult to use or to use python types